server/buildbuddy_server/buildbuddy_server.go

```diff
@@ -142,6 +143,20 @@ func (s *BuildBuddyServer) DeleteInvocation(ctx context.Context, req *inpb.Delet
 	return &inpb.DeleteInvocationResponse{}, nil
 }
 
+func (s *BuildBuddyServer) CancelInvocation(ctx context.Context, req *inpb.CancelInvocationRequest) (*inpb.CancelInvocationResponse, error) {
+	err := s.authorizeInvocationWrite(ctx, req.InvocationId)
+	if err != nil {
+		return nil, err
+	}
+
+	res := s.env.GetRemoteExecutionService()
```

Comment:
Remote execution is an optional (enterprise) feature which means GetRemoteExecutionService could theoretically be nil. It's unlikely CancelInvocation would be invoked, but we should be defensive here and check whether it's nil.

Reply:
Done